sktime/sindy/sindy.py

```diff
@@ -0,0 +1,526 @@
+from typing import Tuple
+
+import numpy as np
+from sklearn.linear_model import LinearRegression, ridge_regression
+from sklearn.metrics import r2_score
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import PolynomialFeatures
+from sklearn.utils.validation import check_X_y
+
+from ..base import Estimator, Model
+from ..numeric import drop_nan_rows
+
+
+class SINDy(Estimator):
+    r"""Learn a dynamical systems model for measurement data using the
+    Sparse Identification of Nonlinear Dynamical Systems (SINDy) method.
+
+    For given measurement data :math:`X`, and a set of library functions evaluated
+    on :math:`X`
+
+    .. math::
+        \Theta(X) = [\theta_1(X), \theta_2(X), \dots, \theta_k(X)],
+
+    SINDy seeks a sparse set of coefficients :math:`\Xi` which satisfies
+
+    .. math::
+        \dot{X} \approx \Theta(X)\Xi.
+
+    The i-th column of this matrix equation gives a differential equation for the
+    i-th measurement variable (i-th column in :math:`X`). For more details see
+    :cite:`sindy-brunton2016sindy`.
+
+    References
+    ----------
+    .. bibliography:: /references.bib
+        :style: unsrt
+        :filter: docname in docnames
+        :keyprefix: sindy-
+    """
+
+    def __init__(self, library=None, optimizer=None, input_features=None):
+        r"""Initialize a new SINDy estimator.
+
+        Parameters
+        ----------
+        library : library object, optional, default=None
+            The candidate feature library, :math:`Theta`.
```

Comment:
There are `:meth:` and `:attr:` directives that you can use instead of `:code:` - in which case sphinx is going to generate links to the methods